api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java

```diff
@@ -41,9 +46,26 @@ private void processMessagesOfType(MessageType messageType) {
                 throw new IllegalArgumentException("No message type processor found in application context for message type: "
                         + messageType + ", there are " + allMessagesOfType.size() + " message(s) in the queue");
             }
-            allMessagesOfType.forEach(messageTypeProcessor::processMessage);
+
+            processMessages(messageTypeProcessor, allMessagesOfType);
+        }
+    }
+
+    private void processMessages(MessageTypeProcessor messageTypeProcessor, List<Message> allMessagesOfType) {
+        Map<MessageStatus, Long> statuses = allMessagesOfType.stream()
+                .map(messageTypeProcessor::processMessage)
+                .peek(messageStatus -> checkForNullMessageStatus(messageTypeProcessor, messageStatus))
+                .filter(Objects::nonNull)
+                .collect(groupingBy(identity(), counting()));
+
+        statuses.forEach((messageStatus, count) -> log.info("Processed {} {} with status {}",
+                count, count == 1 ? "message" : "messages", messageStatus.name()));
+    }
+
+    private void checkForNullMessageStatus(MessageTypeProcessor messageTypeProcessor, MessageStatus messageStatus) {
```

Comment:
Assuming this is not going to do any more than log a message `logNullMessageStatus` might be a better name.

Reply:
Done.